assayist/processor/loose_artifact_analyzer.py

```diff
@@ -38,123 +43,234 @@ def run(self):
+            source_embedded_artifacts.append(artifact)
+
+        # Now examine the build artifacts.
+        for archive, path_to_archive in self.unpacked_archives():
+            # Assume that the artifact being analyzed was created by the main analyzer
+            original_artifact = content.Artifact.nodes.get(filename=archive)
+            # Assume that every artifact found in the source is embedded in every built artifact.
+            for source_artifact in source_embedded_artifacts:
+                original_artifact.embedded_artifacts.connect(source_artifact)
+
+            for loose_artifact in self.files_to_examine(path_to_archive):
+                relative_filepath = os.path.relpath(loose_artifact, path_to_archive)
+
+                try:
+                    artifact = self.local_lookup(loose_artifact)
+                except FileNotFoundError:
+                    # There are two potential causes here, both with symlinks:
+                    # 1) There is a symlink that points to a file in a different
+                    #    layer of the container.
+                    # 2) It was a symlink to something we already analyzed and
+                    #    claimed.
+                    #
+                    # Either way I don't think we really care. If it's already
+                    # claimed then we've already established the link to this
+                    # artifact. If it's referenceing something on a different
+                    # layer of the container then we'll find it when we analyse
+                    # that build (and that's the layer that needs to be respun
+                    # anyway, since that's what contains the actual thing).
+                    # Let's just claim the file and move on.
+                    log.warning(f'Skipping already-claimed symlink in {archive}: '
+                                f'{relative_filepath}')
+                    self.claim_file(path_to_archive, relative_filepath)
+                    continue
+
+                # If we find it locally don't bother asking Koji about it again.
+                if artifact:
+                    self.conditional_connect(original_artifact.embedded_artifacts, artifact)
+                    self.claim_file(path_to_archive, relative_filepath)
+                    continue
+
+                # Add the file to the batch of things to process. If this happens to
+                # trigger a batch execution, handle the resulting Artifacts.
+                for artifact in self.add_to_and_maybe_execute_batch(loose_artifact,
+                                                                    path_to_archive):
+                    self.conditional_connect(original_artifact.embedded_artifacts, artifact)
+
+            # Wrap up any in-progress batch before moving on to the next archive.
+            for artifact in self.execute_batch_and_return_artifacts():
+                self.conditional_connect(original_artifact.embedded_artifacts, artifact)
```

Comment:
It is automatically set back to False when you actually execute `multiCall()`. That's standard koji lib behavior, so I don't think we need to do anything special to account for it.

Reply:
But you set it to `True` at the end of `execute_batch_and_return_artifacts()` so after the last iteration, it will remain `True` when `run()` finishes executing.

Reply:
There, now we only set `multicall=True` at the beginning of a new batch when we are guaranteed to call `multiCall()` at some point in the future.